apps/server/lib/lexical/server/transport/std_io.ex

```diff
@@ -59,44 +61,57 @@ defmodule Lexical.Server.Transport.StdIO do
       "\n" ->
         headers = parse_headers(buffer)
 
-        with {:ok, content_length} <-
-               header_value(headers, "content-length", &String.to_integer/1),
+        with {:ok, content_length} <- content_length(headers),
              {:ok, data} <- read_body(device, content_length),
              {:ok, message} <- JsonRpc.decode(data) do
           callback.(message)
+        else
+          {:error, :empty_response} ->
+            :noop
+
+          {:error, reason} ->
+            Logger.critical("read protocol message: #{inspect(reason)}")
         end
 
         loop([], device, callback)
 
       :eof ->
+        Logger.critical("stdio received :eof, server will stop.")
         maybe_stop()
 
       line ->
         loop([line | buffer], device, callback)
     end
   end
 
-  defp parse_headers(headers) do
-    Enum.map(headers, &parse_header/1)
-  end
+  defp content_length(headers) do
+    case List.keyfind(headers, "content-length", 0) do
+      {_, len_str} ->
+        {:ok, String.to_integer(len_str)}
 
-  defp header_value(headers, header_name, converter) do
-    case List.keyfind(headers, header_name, 0) do
-      nil -> :error
-      {_, value} -> {:ok, converter.(value)}
+      nil ->
+        {:error, :no_content_length_header}
```

Comment:
one more nitpick

```elixir 
with {_, len_str} <- List.keyfind(headers, "content-length", 0),
       {content_length, ""} <- Integer.parse(len_str) do
  {:ok, content_length}
else _ -> 
  {:error, :no_content_length_header}
end
```

Reply:
(String.to_integer raises)

Reply:
That's what was there - could rewrite it like this though:
```elixir
defp content_length(headers) do
  with {:ok, len_str} <- find_header(headers, "content-length"),
       {:ok, int} <- parse_length(len_str) do
    {:ok, int}
  end
end

defp find_header(headers, name) do
  case List.keyfind(headers, name, 0) do
    {_, len_str} -> {:ok, len_str}
    nil -> {:error, {:header_not_found, name}}
  end
end

defp parse_length(len_str) when is_binary(len_str) do
  case Integer.parse(len_str) do
    {int, ""} -> {:ok, int}
    :error -> {:error, {:cant_parse_length, len_str}}
  end
end
```

Reply:
this is fine

Reply:
Added